packages/editor/src/components/block-list/block.js

```diff
@@ -131,7 +133,12 @@ export class BlockListBlock extends Component {
 		}
 
 		// Find all tabbables within node.
-		const textInputs = focus.tabbable.find( this.node ).filter( isTextField );
+		const innerBlocks = this.node.querySelector( '.editor-inner-blocks' );
```

Comment:
This is a future bug waiting to happen.

Reply:
How would you solve it?

Reply:
I worked my way back to questioning whether we need it at all in following comments. We can revisit if still then necessary.

Reply:
We have an `isInSameBlock` function in `editor/utils/dom.js` which could work well for this.

Reply:
@aduth It won't work for the column's block appender. As it's in the same block but we don't want to focus it.

Reply:
Seems like a problem there is that we consider the default appender as a valid text field to target? For what reason is it `<input type="text">`? Should we exclude `readOnly` from auto-focus?

Reply:
I'd be content if we at least abstract this to some common utility in `editor/utils/dom.js`. The file was created after this same sort of bugginess crept up repeatedly, to have a single point of updates for DOM inspection functions.

e.g. `isInSameRootClientId`

Reply:
The appender is an input to allow the writing flow navigation (we discussed this at some point :) I think).

I extracted to a dom utility.